app/cdap/components/Connections/Browser/GenericBrowser/index.tsx

```diff
@@ -116,7 +119,11 @@ export function GenericBrowser({ selectedConnection }) {
     const { name: entityName, canBrowse } = entity;
     if (!canBrowse) {
       setLoading(true);
-      return onCreateWorkspace(entity);
+      getEntitySpec(entity);
```

Comment:
Question: Does this function have side effects? If so, how would you feel about renaming it away from `get...`, eg perhaps `retrieveEntitySpec` or `loadEntitySpec`? should it be duplicated on this line vs the one below?